plugins/net_plugin/net_plugin.cpp

```diff
@@ -3714,7 +3727,7 @@ namespace eosio {
                      ("cid", cid)("n", ptr->block_num())("id", id.str().substr(8,16)));
          }
          if( exception || !obh) {
-            if (!obh) {
+            if (!exception && !obh) {
```

Comment:
This condition `!exception && !obh` is confusing with the `exception || !obh` above.

Reply:
True. I added a `bool unlinkable` which makes it clearer.